core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -113,10 +113,11 @@ internal class CacheFallbackOperations
 #endif
         }
 
-        public static Dictionary<String, AdalUserInfo> GetAllAdalUsersForMsal(ILegacyCachePersistance legacyCachePersistance,
-            ISet<string> environments, string clientId)
+        public static Tuple<Dictionary<String, AdalUserInfo>, List<AdalUserInfo>> GetAllAdalUsersForMsal
```

Comment:
I suggest you use the C# 7 style tuples or at least add a comment on this method to explain what it returns. 

https://blogs.msdn.microsoft.com/mazhou/2017/05/26/c-7-series-part-1-value-tuples/

Reply:
I believe Petro couln't get that to work in his previous PR for some reason. It might be because this is not yet available for all platforms?

Reply:
Ah ok, makes sense. net45 probably does not support this (you might need to take a dependency on a NuGet package for this... not worth the trouble).